### ISSUE TITLE
Opacity updated when editor loses focus (blurred)

### ISSUE THREAD

**Issue by hhhapz** [glacambre/firenvim#1100]
At the moment, when the editor is blurred, the element gets 0.5 opacity.  Is
there a way to disable this? I am personally not a fan of it.  At the moment, I
have just forked the repository and commented it out, however, is a reason
the opacity is updated?

**Comment by glacambre** [comment]
The opacity is updated in order to tell the user that the frame isn't focused anymore. This is important because some people use vim-like extensions (e.g. Tridactyl) to navigate the web and could accidentally close their tab by pressing `d` when the page is focused instead of Firenvim.
I'm not against making this configurable but it should probably be more than just a boolean toggle and allow people to completely change the way firenvim focus/blur is communicated to them.

**Comment by hhhapz** [comment]
I understand! Any suggestions/ideas on how you think we could implement this? I'm honestly not actually very familiar with how extensions can get settings so I might not be able to implement this one.

**Comment by glacambre** [comment]
>Any suggestions/ideas on how you think we could implement this?

Before going into implementation details we need to design this feature :). Do we want to keep using CSS or start using Vim highlights? Do we want to allow setting arbitrary styles for the outer frame? I don't know what the answer to these questions is right now and there probably are more questions we should ask ourselves. Unfortunately I'm a bit too busy right to think about these and I'm not sure how to guide you through discovering the various constraints the webextension platform puts on extensions.

If you want to purely experiment with setting and getting options, you can copy what happens in [background.ts](https://github.com/glacambre/firenvim/blob/565061ec6d437c5ce90485e2d99cd8bedab1cc0b/src/background.ts#L162) for option creation and see how [getConfForUrl works in frame.ts](https://github.com/glacambre/firenvim/blob/565061ec6d437c5ce90485e2d99cd8bedab1cc0b/src/frame.ts#L44).

**Comment by marcelarie** [comment]
IMO a border when the buffer is focused will be better. Its there any way to disable the transparency on blur?

**Comment by glacambre** [comment]
>Its there any way to disable the transparency on blur?

No, there is no way to do this currently. I understand the need, though.

**Comment by XDTZ** [comment]
I made a small donation because I wish this would be implemented someday.
